nixos/modules/virtualisation/containers.nix

```diff
@@ -115,8 +121,12 @@ in
     environment.etc."containers/containers.conf".text = ''
       [network]
       cni_plugin_dirs = ["${pkgs.cni-plugins}/bin/"]
-
-    '' + cfg.containersConf.extraConfig;
+    '' + (if cfg.enableOciSeccompBpfHook then ''
+      [engine]
+      hooks_dir = [
+        "${config.boot.kernelPackages.oci-seccomp-bpf-hook}",
+      ]
+    '' else "") + cfg.containersConf.extraConfig;
```

Comment:
Could use `optionalString` to avoid the `if then ... else ""`.

```nix
environment.etc."containers/containers.conf".text = ''
  [network]
  cni_plugin_dirs = ["${pkgs.cni-plugins}/bin/"]

  ${lib.optionalString (cfg.enableOciSeccompBpfHook == true) ''
  [engine]
  hooks_dir = [
    "${config.boot.kernelPackages.oci-seccomp-bpf-hook}",
  ]
  ''}

'' + cfg.containersConf.extraConfig;
```

Reply:
Thanks for the hint, fixed as suggested :+1: